modules/messages/client/components/Inbox.component.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/messages.api';
+import Avatar from '@/modules/users/client/components/Avatar.component';
+import Activate from '@/modules/users/client/components/Activate';
+import { $broadcast, eventTrack } from '@/modules/core/client/services/angular-compat';
+import TimeAgo from '@/modules/core/client/components/TimeAgo';
+
+const userType = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  displayName: PropTypes.string.isRequired,
+});
+
+export default function Inbox({ user }) {
+  if (!user.public) {
+    return (
+      <section className="container-spacer">
+        <Activate/>
+      </section>
+    );
+  }
+  const { t } = useTranslation('messages');
+
+  const [nextParams, setNextParams] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [threads, setThreads] = useState([]);
+
+  const hasMore = Boolean(nextParams);
+
+  async function fetchThreads(next = false) {
+    setIsFetching(true);
+    try {
+      if (next) {
+        eventTrack('inbox-pagination', {
+          category: 'messages.inbox',
+          label: 'Inbox page ' + nextParams,
```

Comment:
Was `nextParams` Object or Integer?

Reply:
It's an object (with page and per page values iirc), I attempted to just replicate it as it was used before (to at least make the `label` consistent with existing values), although I didn't think about it too much.

How is that label used? Perhaps just showing the page number (and ignoring the per page value) would make more sense...

Reply:
The label just allows seeing how far in history people typically browse but I haven't been interested in that for long time so ok to drop the label.

Reply:
I changed it to just have the page number. Perhaps we can rip out lots of those analytics calls at some point, but I'll keep it in for now, seeing as it was already there.